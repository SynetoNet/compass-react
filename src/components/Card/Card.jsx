import React from "react";
import PropTypes from "prop-types";
import BSCard from "react-bootstrap/Card";

import "./Card.scss";

const Card = ({ title, content, dropdown, icon, ...props }) => {
  return (
    <BSCard {...props}>
      <div className="cardIconContainer">{icon}</div>
      <BSCard.Body>
        <BSCard.Title>{title}</BSCard.Title>
        <BSCard.Text as={"div"}>{content}</BSCard.Text>
      </BSCard.Body>
      
      {dropdown ? <div className="cardActions">{dropdown}</div> : null}
    </BSCard>
  );
};

Card.propTypes = {
  title: PropTypes.string,
  content: PropTypes.node,
  dropdown: PropTypes.node,
  icon: PropTypes.node
};

Card.defaultProps = {};

export default Card;
