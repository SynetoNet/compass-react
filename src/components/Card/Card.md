#### **Card with Icon component**

```jsx
import { Icon } from "../../index.js";
<Card
  icon={<Icon name="fas fa-user fa-md" />}
  title={"Icon Card Title"}
  content={
    " Some quick example text to build on the card title and make up the bulk of the card's content."
  }
/>;
```

#### **Card with img **

In this case, you have to specify the external svg url.

```jsx
<Card
  icon={
    <img width="100px" height="100px" src="https://svgshare.com/i/EDi.svg" />
  }
  content={
    " Some quick example text to build on the card title and make up the bulk of the card's content."
  }
  title={"SVG Card Title"}
/>
```

#### **Card with actions dropdown**

```jsx
import { Badge, Dropdown } from "../../index.js";
<Card
  icon={
    <img width="100px" height="100px" src="https://svgshare.com/i/EDi.svg" />
  }
  content={
    " Some quick example text to build on the card title and make up the bulk of the card's content."
  }
  title={"SVG Card Title"}
  dropdown={
    <Dropdown
      label={"Actions"}
      onSelect={eventKey => console.log(eventKey)}
      variant="tertiary"
      items={[
        {
          label: "Action",
          eventKey: "action-key",
          onSelect: () => console.log("Additional handler")
        },
        {
          label: "No event key",
          onSelect: () => console.log("eventKey is the label")
        },
        {
          label: "No additional handler"
        },
        {
          label: "Some very long action name"
        }
      ]}
    />
  }
/>;
```
