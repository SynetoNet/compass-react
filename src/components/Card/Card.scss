@import "../../base/colors";
@import "../../base/variables";

.card {
  flex-direction: row;
}

.card-icon-container {
  display: flex;
  align-items: center;

  & img,
  & i {
    font-size: 32px;
    padding: 0 16px;
  }
}

.card-actions {
  display: flex;
  margin-top: 18px;
  margin-right: 18px;
  font-weight: bold;
  font-size: 32px;

  & button {
    color: $neutral-100;

    &:hover {
      background-color: $neutral-24 !important;
      color: $neutral-90 !important;
    }

    &:focus {
      box-shadow: none;

      &:active {
        background-color: $neutral-24 !important;
        color: $neutral-90 !important;
      }
    }
  }
}
