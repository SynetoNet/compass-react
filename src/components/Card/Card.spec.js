import React from "react";
import Card from "./Card";
import { Icon } from "../../index.js";
import renderer from "react-test-renderer";

test("renders default Badge", () => {
  const component = renderer.create(
    <Card
      icon={<Icon name="fas fa-user fa-md" />}
      title={"Icon Card Title"}
      content={
        " Some quick example text to build on the card title and make up the bulk of the card's content."
      }
    />
  );
  expect(component.toJSON()).toMatchSnapshot();
});
