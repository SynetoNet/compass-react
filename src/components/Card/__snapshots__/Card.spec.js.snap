// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default Badge 1`] = `
<div
  className="card"
>
  <div
    className="cardIconContainer"
  >
    <i
      className="fas fa-user fa-md fa-md"
      style={
        Object {
          "color": undefined,
        }
      }
    />
  </div>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Icon Card Title
    </div>
    <div
      className="card-text"
    >
       Some quick example text to build on the card title and make up the bulk of the card's content.
    </div>
  </div>
</div>
`;
